src/geoaxis.jl

```diff
@@ -0,0 +1,107 @@
+"""
+    GeoAxis(args...; kwargs...) → ax
+Create a new axis instance `ax` that is a modified `Axis` of the Makie.jl ecosystem.
+All Makie.jl plotting functions work directly on `GeoAxis`, e.g., `scatter!(ax, x, y)`.
+
+`GeoAxis` is appropriate for geospatial plotting because it automatically transforms all
+plotted data given a user-defined map projection. See keyword arguments below and examples
+in the online documentation. Longitude and latitude values in GeoMakie.jl are always
+assumed to be **in degrees**.
+
+In the call signature, `args...` is a standard figure location, e.g., `fig[1,1]` as given in
+`Axis`. The keyword arguments decide the geospatial projection:
+
+* `source = "+proj=longlat +datum=WGS84", dest = "+proj=eqearth"`: These two keywords
+  configure the map projection to be used for the given field using Proj4.jl.
+  See also online the section [Changing central longitude](@ref) for data that may not
+  span the (expected by default) longitude range from -180 to 180.
+* `transformation = Proj4.Transformation(source, dest, always_xy=true)`: Instead of
+  `source, dest` you can directly use the Proj4.jl package to define the projection.
+  lines!(ax, GeoMakie.coastlines(); coastkwargs...)
+* `lonticks = -180:60:180, latticks = -90:30:90` ticks for the longitude and latitude
+  dimensions. The grid lines of the axis are also spanning these tick values.
+* `hidespines = true` Hide the axis spines (rectangle surrounding the axis).
+
+## Example
+```julia
+using GeoMakie, GLMakie
+
+lons = -180:180
+lats = -90:90
+field = [exp(cosd(l)) + 3(y/90) for l in lons, y in lats]
+
+# Plot coastlines
+coastplot = lines!(ax, GeoMakie.coastlines(); color = :black, overdraw = true, coastkwargs...)
+translate!(coastplot, 0, 0, 99) # ensure they are on top of other plotted elements
+
+# Surface example
+fig = Figure()
+ax = GeoAxis(fig[1,1])
+surface!(ax, lons, lats, field)
+display(fig)
```

Comment:
I get the following error:
```
Error in callback:
MethodError: no method matching inverse_transform(::Makie.PointTrans{2, Proj4.Transformation})
Closest candidates are:
  inverse_transform(::Tuple) at ~/.julia/packages/Makie/gQOQF/src/layouting/transformation.jl:360
  inverse_transform(::typeof(identity)) at ~/.julia/packages/Makie/gQOQF/src/layouting/transformation.jl:354
  inverse_transform(::typeof(log10)) at ~/.julia/packages/Makie/gQOQF/src/layouting/transformation.jl:355
  ...
Stacktrace:
  [1] process_interaction(r::Makie.MakieLayout.RectangleZoom, event::MouseEvent, ax::Axis)
    @ Makie.MakieLayout ~/.julia/packages/Makie/gQOQF/src/makielayout/interactions.jl:150
  [2] (::Makie.MakieLayout.var"#process_event#312"{Axis})(event::MouseEvent)
    @ Makie.MakieLayout ~/.julia/packages/Makie/gQOQF/src/makielayout/layoutables/axis.jl:410
  [3] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
  [4] invokelatest
    @ ./essentials.jl:714 [inlined]
  [5] (::Makie.PrioCallback)(val::MouseEvent)
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:17
  [6] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
  [7] invokelatest
    @ ./essentials.jl:714 [inlined]
  [8] notify(observable::Makie.PriorityObservable{MouseEvent})
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:85
  [9] setindex!
    @ ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:79 [inlined]
 [10] (::Makie.MakieLayout.var"#99#101"{Scene, Makie.MakieLayout.var"#is_mouse_over_relevant_area#96"{Scene, Tuple{}}, Base.RefValue{Float64}, Base.RefValue{Bool}, Base.RefValue{Bool}, Base.RefValue{Union{Nothing, Makie.Mouse.Button}}, Base.RefValue{Bool}, Base.RefValue{GeometryBasics.Point{2, Float32}}, Base.RefValue{GeometryBasics.Point{2, Float32}}, Base.RefValue{Makie.Mouse.Action}, Makie.PriorityObservable{MouseEvent}, Module})(mp::Tuple{Float64, Float64})
    @ Makie.MakieLayout ~/.julia/packages/Makie/gQOQF/src/makielayout/mousestatemachine.jl:211
 [11] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [12] invokelatest
    @ ./essentials.jl:714 [inlined]
 [13] (::Makie.PrioCallback)(val::Tuple{Float64, Float64})
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:17
 [14] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [15] invokelatest
    @ ./essentials.jl:714 [inlined]
 [16] notify(observable::Makie.PriorityObservable{Tuple{Float64, Float64}})
    @ Makie ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:85
 [17] setindex!
    @ ~/.julia/packages/Makie/gQOQF/src/interaction/PriorityObservable.jl:79 [inlined]
 [18] (::GLMakie.var"#69#70"{Events, GLFW.Window})(#unused#::Nothing)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/events.jl:6
 [19] #invokelatest#2
    @ ./essentials.jl:716 [inlined]
 [20] invokelatest
    @ ./essentials.jl:714 [inlined]
 [21] notify
    @ ~/.julia/packages/Observables/OFj0u/src/Observables.jl:88 [inlined]
 [22] setindex!
    @ ~/.julia/packages/Observables/OFj0u/src/Observables.jl:248 [inlined]
 [23] fps_renderloop(screen::GLMakie.Screen, framerate::Float64)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/rendering.jl:22
 [24] renderloop(screen::GLMakie.Screen; framerate::Float64)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/rendering.jl:48
 [25] renderloop(screen::GLMakie.Screen)
    @ GLMakie ~/.julia/packages/GLMakie/pFGSp/src/rendering.jl:41
 [26] (::GLMakie.var"#50#52"{GLMakie.Screen})()
    @ GLMakie ./task.jl:423
```

Reply:
Ah, that appears to be https://github.com/JuliaPlots/GeoMakie.jl/issues/77

Reply:
Yeah this is known but Simon said its solved on master.

Reply:
master of Makie?